src/app/util/attribute-storage.cpp

```diff
@@ -71,6 +71,10 @@ EmberAfDefinedEndpoint emAfEndpoints[MAX_ENDPOINT_COUNT];
 uint8_t attributeData[ACTUAL_ATTRIBUTE_SIZE];
 
 namespace {
+#if (!defined(DEFAULT_VALUE_SIZE_IN_DEFAULT_OR_MAX_MIN_STRUCT)) || (DEFAULT_VALUE_SIZE_IN_DEFAULT_OR_MAX_MIN_STRUCT == 0)
+#define DEFAULT_VALUE_SIZE_IN_DEFAULT_OR_MAX_MIN_STRUCT 2
```

Comment:
@mrjerryjohns 
The reason behind `define DEFAULT_VALUE_SIZE_IN_DEFAULT_OR_MAX_MIN_STRUCT x` and not just using `sizeof(EmberAfDefaultOrMinMaxAttributeValue)` is because I don't know if there is code outside of this repo will not be updated their `endpoint_config.h` header file when this PR lands. Those other legacy files need to still work right? I can't just rely on `sizeof(EmberAfDefaultOrMinMaxAttributeValue)` since that will always give me 4 on legacy targets that have not updated their `endpoint_config.h` that are 32-bit address (and will for sure be 4 bytes after this change is in).

Reply:
Are we sure that the practice of regenerating `endpoint_config.h` always happens? I don't know what the general process is outside, so I was trying to stick with the safe assumption that they don't always regenerate the files while I was writing this.

If we can assume that regenerating files always happens and they also use the zapt file we have included in the repo (ei they don't use some custom one) than I can likely take Jerry's suggestion. If those are not safe assumption and I apply Jerry's comments, than the fallout would be for attributes that are size 3 and 4 bytes will use a pointer as the value

Reply:
If it's not happening, it's been broken by all sorts of things recently (e.g. the changes to the device list format).